### ISSUE TITLE
Change recommended star rating formula to be more in line with how performance points works

### ISSUE THREAD

**Issue by creatorl2** [ppy/osu-web#9020]
Currently, the recommended difficulty on the beatmaps page is entirely inaccurate at extreme ends of the skill spectrum. For example, players below ~2000pp who are recommended maps that are an entire 0.5* below what they are capable of playing, or top ~100 players who are recommended maps that are an entire 1* above their skill level.

I propose changing the recommended difficulty formula to (_totalPp_ * 0.85 / 20) ^ (1 / 3), which basically just takes the cube root of a fraction of your total pp. This follows how pp actually correlates with star rating, as much of the pp a map gives is quite literally star rating to the power of 3. The *0.85 is to keep values similar to where they were before for average skill level (4000pp-7000pp) players, where recommended star rating is actually close to what players at that skill level play. What this formula does can be seen here in this desmos: https://www.desmos.com/calculator/dsj0oubocw
 
This change overall would make recommended star rating better by it being more accurate to what players at a given pp skill level play, and how pp actually functions.

**Comment by LittleEndu** [comment]
I assume `(pp ^ 0.333) * 0.35` is almost the same as proposed algorithm but easier to express. Otherwise seems like a good proposal.

**Comment by Zyfarok** [comment]
> I assume `(pp ^ 0.333) * 0.35` is almost the same as proposed algorithm but easier to express.

Except that the 0.35 looks like a magic number while the 0.85 / 20 can be at least partially explained : the total weight of all maps on a profile soft-caps at 20 (2000%), so dividing by 20 gives back the weighted average of top plays pp.

**Comment by Zyfarok** [comment]
> The soft cap is actually around 19.88

No. 19.88 is if you only count your top 100 but all your scores give you pp even if not in your top 100. Also it's a soft cap because you'll never reach it but can get any arbitrarily close to it. If the weights are changed to something different than the current `(19/20)^i` then the 20 could trivially be changed too.

I agree the 0.85 is still "magic" and moving it outside of the cube-root would be fine (`((pp / 20) ^ (1/3)) * 0.95`) but it's easily understandable that it's a slight correction factor that has to be close to 1.

**Comment by creatorl2** [comment]
x_x ok really sorry lol

**Comment by creatorl2** [comment]
but yeah the 0.85 is just a correction factor as i said, its based of a survey i took that leads me to believe that people in the 6* range are fine with their current recommended difficulty

**Comment by Givikap120** [comment]
I would propose this kind of mechanism:

There are 3 values:
Upper bound = player pp top play * 1.2 + 10pp
Best match = arithmetic mean of player pp top play and player profile total pp / 20 (to represent average top play)
Lower bound = player top100 play, 0pp if player have less than 100 scores

Then it checks through difficulties of the map and checks the amount of pp for FC with active mods and recommended accuracy. When there is at least one difficulty between lower and upper bound - it highlights the difficulty closest to best match.

recommended accuracy - return value of the function that shows expected accuracy based on the map's OD and player skill level represented in profile pp
